src/library_openal.js

```diff
@@ -103,898 +193,3990 @@ var LibraryOpenAL = {
+    case 0x1102:  /* AL_FORMAT_STEREO8 */
+      outputChannelCount = 2;
+      break;
+    default:
+#if OPENAL_DEBUG
+      console.error('alcCaptureOpenDevice() with unsupported format ' + format);
+#endif
+      AL.alcErr = 0xA004 /* ALC_INVALID_VALUE */;
+      return 0;
+    }
+
+    function newF32Array(cap) { return new Float32Array(cap);}
+    function newI16Array(cap) { return new Int16Array(cap);  }
+    function newU8Array(cap)  { return new Uint8Array(cap);  }
+
+    var requestedSampleType;
+    var newSampleArray;
+
+    switch (format) {
+    case 0x10010: /* AL_FORMAT_MONO_FLOAT32 */
+    case 0x10011: /* AL_FORMAT_STEREO_FLOAT32 */
+      requestedSampleType = 'f32';
+      newSampleArray = newF32Array;
+      break;
+    case 0x1101:  /* AL_FORMAT_MONO16 */
+    case 0x1103:  /* AL_FORMAT_STEREO16 */
+      requestedSampleType = 'i16';
+      newSampleArray = newI16Array;
+      break;
+    case 0x1100:  /* AL_FORMAT_MONO8 */
+    case 0x1102:  /* AL_FORMAT_STEREO8 */
+      requestedSampleType = 'u8';
+      newSampleArray = newU8Array;
+      break;
+    }
+
+    var buffers = [];
+    try {
+      for (var chan=0; chan < outputChannelCount; ++chan) {
+        buffers[chan] = newSampleArray(bufferFrameCapacity);
+      }
+    } catch(e) {
+#if OPENAL_DEBUG
+      console.error('alcCaptureOpenDevice() failed to allocate internal buffers (is bufferSize low enough?): ' + e);
+#endif
+      AL.alcErr = 0xA005 /* ALC_OUT_OF_MEMORY */;
+      return 0;
+    }
+
+
+    // What we'll place into the `AL.captures` array in the end,
+    // declared here for closures to access it
+    var newCapture = {
+      audioCtx: AL.sharedCaptureAudioCtx,
+      deviceName: resolvedDeviceName,
+      requestedSampleRate: requestedSampleRate,
+      requestedSampleType: requestedSampleType,
+      outputChannelCount: outputChannelCount,
+      inputChannelCount: null, // Not known until the getUserMedia() promise resolves
+      mediaStreamError: null, // Used by other functions to return early and report an error.
+      mediaStreamSourceNode: null,
+      // Either one, or none of the below two, is active.
+      mergerNode: null,
+      splitterNode: null,
+      scriptProcessorNode: null,
+      isCapturing: false,
+      buffers: buffers,
+      get bufferFrameCapacity() {
+        return buffers[0].length;
+      },
+      capturePlayhead: 0, // current write position, in sample frames
+      capturedFrameCount: 0
+    };
+
+    // Preparing for getUserMedia()
+
+    var onError = function(mediaStreamError) {
+      newCapture.mediaStreamError = mediaStreamError;
+#if OPENAL_DEBUG
+      console.error('navigator.getUserMedia() errored with: ' + mediaStreamError);
+#endif
+    };
+    var onSuccess = function(mediaStream) {
+      newCapture.mediaStreamSourceNode = newCapture.audioCtx.createMediaStreamSource(mediaStream);
+
+      var inputChannelCount = 1;
+      switch(newCapture.mediaStreamSourceNode.channelCountMode) {

```

Comment:
Missing s. It should be `AL.buffers[0]` instead of `AL.buffer[0]`

Reply:
Good catch. Fixed.